src/main/webapp/components/query-selector/query-selector.js

```diff
@@ -31,12 +35,38 @@ const useOpenClose = props => {
   return [anchorEl, open, handleOpen, handleClose]
 }
 
+const defaultSearchForms = {
+  basic: BasicSearchQueryBuilder,
+  advanced: AdvancedSearchQueryBuilder,
+}
+
+const defaultSearchFormLabels = {
+  basic: 'Basic Search',
+  advanced: 'Advanced Search',
+}
+
 const QueryCard = props => {
-  const { onSearch, QueryEditor, onChange } = props
+  const { onSearch, QueryEditor, onChange, query } = props
   const [anchorEl, open, handleOpen, handleClose] = useOpenClose(props)
   const [{ active: isDrawing }] = useDrawInterface()
   const [wasDrawing, setWasDrawing] = useState(false)
   const drawAnchorEl = useRef(null)
+  const [queryBuilder, setQueryBuilder] = React.useState(
+    defaultSearchForms.hasOwnProperty(query.type) ? query.type : 'advanced'
+  )
+
+  const defaultSearchFormInteractions = Object.keys(defaultSearchForms).map(
+    key => {
+      return (
+        <MenuItem key={key} onClick={() => setQueryBuilder(key)}>
```

Comment:
Should we clear the query each time we change query builders? This is what is done in ddf, I'm guessing because the different searches do not have the same available filters

Reply:
I think we should clear the filterTree at least for now, because there are certain filters that won't work in both builders like you said. We would want to at least preserve the id either way though